lib/stat_tracker.rb

```diff
@@ -23,16 +23,33 @@ def initialize(game_path, team_path, game_teams_path)
     @games = @game_collection.all_games
   end
 
+# Game Statistics Tests - Helper Methods #
+  def seasons 
+    @games.map {|game| game.season}.uniq 
+  end
 
-  # Game Statistics #
-  # Helper Methods #
   def total_goals_per_game
     @games.reduce({}) do |ids_to_scores, game|
       ids_to_scores[game.game_id] = game.away_goals + game.home_goals
       ids_to_scores
     end
   end
 
+  def array_of_total_goals_per_season
+    @games.reduce(Hash.new { |h, k| h[k] = [] }) do |result, game|
+      result[game.season] << game.away_goals + game.home_goals
+      result
+    end
+  end
+
+  def total_goals_per_season
+    result = {}
+    array_of_total_goals_per_season.each do |season, goals|
+      result[season] = goals.sum
+    end
+    result
+  end
```

Comment:
I think these two methods could just be one like this:

```
def total_goals_per_season
    result = Hash.new(0)
    @games.each do |game|
      result[game.season] += game.away_goals + game.home_goals
    end
    result
  end
```

Unless you will use the breakdown given by `array_of_total_goals_per_season`. The above could also be refactored using reduce.

Reply:
@guardiac yesssss, thank you! I've refactored this down to one method using .reduce.  I don't have a need for the breakdown from `array_of_total_goals_per_season`.